app/src/main/java/org/fossasia/openevent/app/core/event/list/EventsListAdapter.java

```diff
@@ -28,24 +29,51 @@
 class EventsListAdapter extends RecyclerView.Adapter<EventsListAdapter.EventRecyclerViewHolder>
     implements StickyRecyclerHeadersAdapter<HeaderViewHolder>, Filterable {
 
-    private final List<Event> events;
+    private List<Event> events;
     private final List<Event> selectedEvents = new ArrayList<>();
 
     private final Bus bus;
-    private final EventsPresenter eventsPresenter;
     private boolean sortByName;
 
     private boolean isLongPressed;
 
-    EventsListAdapter(List<Event> events, Bus bus, EventsPresenter eventsPresenter) {
+    private final EventListFragment fragment;
+
+    EventsListAdapter(List<Event> events, Bus bus, EventListFragment fragment) {
         this.events = events;
         this.bus = bus;
-        this.eventsPresenter = eventsPresenter;
+        this.fragment = fragment;
     }
 
-    public void categorizeEvents() {
+    public void updateList(List<Event> newEvents) {
+        if (events == null) {
+            events = newEvents;
+            notifyItemRangeInserted(0, newEvents.size());
+        } else {
+            DiffUtil.DiffResult diffResult = DiffUtil.calculateDiff(new DiffUtil.Callback() {
+                @Override
+                public int getOldListSize() {
+                    return events == null ? 0 : events.size();
```

Comment:
events will never be null in this case. You're already doing the null check in the `if` condition.